.github/workflows/infrastructure.yml

```diff
@@ -23,150 +23,418 @@
+
+      - name: Production 💫
+        if: ${{ 'prod' == env.target }}
+        run: echo "type=Production" >> $GITHUB_ENV
+
+      - name: Tags 🏷️
+        run: echo tags='Environment=${{ env.type }}' \
+          'Product=${{ env.product }}' \
+          'Team=development' >> $GITHUB_ENV
+
+      - name: Login 🔐
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: Azure defaults ✨
+        uses: Azure/cli@1.0.4
+        with:
+          inlineScript: |
+            # Basic
+            az configure --defaults location=${{ vars.REGION }}
+            az configure --defaults group=rg-${{ env.product }}-${{ env.target }}-001
+      
+      - name: MDM - Product ➕
+        uses: Azure/cli@1.0.4
+        with:
+          inlineScript: |
+            az apim product create \
+            --service-name apim-${{ env.environment }}-${{ env.target }}-001 \
+            --product-name apim-${{ env.product }}-mdm \
+            --description 'Master database management endpoints' \
+            --state published \
+            --approval-required true \
+            --subscription-required true
+      
+      - name: TFS - Product ➕
+        uses: Azure/cli@1.0.4
+        with:
+          inlineScript: |
+            az apim product create \
+            --service-name apim-${{ env.environment }}-${{ env.target }}-001 \
+            --product-name apim-${{ env.product }}-tfs \
+            --description 'Trade finance system endpoints' \
+            --state published \
+            --approval-required true \
+            --subscription-required true
+
+      - name: eStore - Product ➕
+        uses: Azure/cli@1.0.4
+        with:
+          inlineScript: |
+            az apim product create \
+            --service-name apim-${{ env.environment }}-${{ env.target }}-001 \
+            --product-name apim-${{ env.product }}-estore \
+            --description 'eStore document management system endpoints' \
+            --state published \
+            --approval-required true \
+            --subscription-required true
+
+      - name: MDM - API import ⬇️
+        uses: Azure/cli@1.0.4
+        with:
+          inlineScript: |
+            az apim api import \
+            --display-name 'MDM' \
+            --description 'Master database management endpoints' \
+            --service-url https://$(az containerapp show --name ca-${{ env.product }}-mdm-${{ env.target }}-001 --query properties.latestRevisionFqdn -o tsv) \
+            --path 'mdm' \
+            --service-name apim-${{ env.environment }}-${{ env.target }}-001 \
+            --specification-format OpenApi \
+            --api-type http \
+            --protocols https \
+            --subscription-required true \
+            --specification-url https://$(az containerapp show --name ca-${{ env.product }}-mdm-${{ env.target }}-001 --query properties.latestRevisionFqdn -o tsv)/openapi/json
```

Comment:
If we're choosing the specification URLs anyway then we don't need to have it at `/openapi/json` anymore, we could keep it at `docs-json`. I think Gabriel only wanted to change it to match one of Azure's expected URLs so that we could import a container app directly

Reply:
I introduced above URL since it does not require any basic authentication also I think it is sets template for any future exports in different format i.e. `/openapi/yaml` and etc.

Reply:
This will probably be for a different PR, but we do need the authentication part 🙂 otherwise anyone can bypass it and execute the codebase 

But we can change it in a separate PR to use an authentication header rather than username/password

Reply:
Only APIM will be able to access the above link, since CA has IP restriction in place. I tried sending the headers and unfortunately headers cannot be send as part of `speification-url`.

Reply:
@abhi-markan, the basic authentication for the endpoint is configured in `src/swagger.ts` with the code
```javascript
  app.use(
    ['/docs', '/docs-json', '/docs-yaml'],
    basicAuth({
      challenge: true,
      users: {
        [configService.get<string>('SWAGGER_USER')]: configService.get<string>('SWAGGER_PASSWORD'),
      },
    }),
  );
```
Can we please update the array to not mention `'/docs-json'` anymore since that URL isn't used anymore? (You could also have done that _instead_ of changing the URL for the JSON docs)

@IgnatG when you said
> we do need the authentication part
> otherwise anyone can bypass it and execute the codebase

Users still need to know a valid API key to be able to use the endpoints. Access to the swagger pages (the UI, the JSON file, and the YAML file) only allows the user to _view the documentation_ of the endpoints. I don't know how sensitive we consider this documentation - is IP restriction enough or do we need authentication as well?

Reply:
Have removed `/docs-json` and `/docs-yaml` and have also introduced `/openapi/yaml` endpoint.
For endpoints documentation anyone can run it locally to extract API documentations.